opal/corelib/struct.rb

```diff
@@ -83,7 +83,15 @@ def [](name)
 
       name = members[name]
     elsif String === name
-      raise NameError.new("no member '#{name}' in struct", name) unless members.include?(name.to_sym)
+      %x{
+        if(!self.$$data.hasOwnProperty(name)) {
+          Opal.Kernel.$raise(
+            Opal.NameError.$new(
+              "no member '" + name + "' in struct", name
+            )
```

Comment:
Unless it's a perfect bottleneck it's better to keep the raise in an interpolation for better consistency with the rest of the codebase